pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1520,25 +1561,27 @@ func TestMergePullSecrets(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			fakeClient := fake.NewFakeClient(test.existingCD...)
+			if test.globalPullSecret != "" {
+				globalPullSecret := createGlobalPullSecretObj(corev1.SecretTypeDockerConfigJson, globalPullSecret, corev1.DockerConfigJsonKey, test.globalPullSecret)
+				test.existingObjs = append(test.existingObjs, globalPullSecret)
+			}
+			fakeClient := fake.NewFakeClient(test.existingObjs...)
```

Comment:
This test needs to register the hive api with the scheme. There needs to be a call to the following somewhere before the fake client is created. In other tests, this is usually done at the top of the function.
```
	apis.AddToScheme(scheme.Scheme)
```

Reply:
This test does not do a reconcile and does need to know all resources as only a single function is getting tested. Let me know if you still think we need to add `apis.AddToScheme(scheme.Scheme)`

Reply:
The test adds resources to the fake client and does gets of resources. Both of those require the types added to the scheme. What do you get when you run the following without adding the api to the scheme?
```
go test ./pkg/controller/clusterdeployment/ -run TestMergePullSecrets
```

Reply:
Got it. Thanks for the explanation.